packages/block-editor/src/hooks/layout.js

```diff
@@ -444,13 +445,21 @@ export const withLayoutStyles = createHigherOrderComponent(
 		);
 
 		if ( ! extraProps ) {
-			return <BlockListBlock { ...props } />;
+			return (
+				<BlockListBlock
+					{ ...props }
+					__unstableLayoutClassNames={
+						layoutClasses.length > 0 ? layoutClasses : undefined
+					}
+				/>
+			);
```

Comment:
The problem seems to be that `useLayoutClasses` doesn't actually check if the block supports layout in order to output the classes (which is fine, that's not its role), so here, we're always passing in layout classes, whether the block supports layout or not. I think the check here should be `blockSupportsLayout ? layoutClasses : undefined`

Reply:
Oh, good catch!

Reply:
For posterity: the unwanted layout classes were appearing on Page List was that it uses `innerBlocksProps`, where those classnames are retrieved from the block edit context and added to the inner blocks wrapper. So this would happen on any block with inner blocks and without layout support.